packages/tab-bar/src/mwc-tab-bar.ts

```diff
@@ -0,0 +1,184 @@
+  private _getTabs() {
+    return this.tabsSlot.assignedNodes({flatten: true}).filter((e: Node) => e instanceof Tab) as Tab[];
+  }
+
+  private _getTab(index) {
+    return this._getTabs()[index];
+  }
+
+  createAdapter() {
+    return {
+      ...super.createAdapter(),
+      scrollTo: (scrollX: number) => this.scrollerElement.scrollToPosition(scrollX),
+      incrementScroll: (scrollXIncrement: number) => this.scrollerElement.incrementScrollPosition(scrollXIncrement),
+      getScrollPosition: () => this.scrollerElement.getScrollPosition(),
+      getScrollContentWidth: () => this.scrollerElement.getScrollContentWidth(),
+      getOffsetWidth: () => this.mdcRoot.offsetWidth,
+      isRTL: () => window.getComputedStyle(this.mdcRoot).getPropertyValue('direction') === 'rtl',
+      setActiveTab: (index: number) => this.mdcFoundation.activateTab(index),
+      activateTabAtIndex: (index: number, clientRect: ClientRect) => {
+        const tab = this._getTab(index);
+        if (tab !== undefined) {
+          tab.activate(clientRect);
+        }
+        this._previousActiveIndex = index;
+      },
+      deactivateTabAtIndex: (index: number) => {
+        const tab = this._getTab(index);
+        if (tab !== undefined) {
+          tab.deactivate()
+        }
+      },
+      focusTabAtIndex: (index: number) => {
+        const tab = this._getTab(index);
+        if (tab !== undefined) {
+          tab.focus();
+        }
+      },
+      // TODO(sorvell): tab may not be able to synchronously answer `computeIndicatorClientRect`
+      // if an update is pending or it has not yet updated. If this is necessary,
+      // LitElement may need a `forceUpdate` method.
+      getTabIndicatorClientRectAtIndex: (index: number) => {
+        const tab = this._getTab(index);
+        return tab !== undefined ? tab.computeIndicatorClientRect() : new DOMRect();
+      },
+      getTabDimensionsAtIndex: (index: number) => {
+        const tab = this._getTab(index);
+        return tab !== undefined ? tab.computeDimensions() :
+            {rootLeft: 0, rootRight: 0, contentLeft: 0, contentRight: 0};
+      },
+      getPreviousActiveTabIndex: () => {
+        return this._previousActiveIndex;
+      },
+      getFocusedTabIndex: () => {
+        const tabElements = this._getTabs();
+        const activeElement = (this as any).getRootNode().activeElement;
+        return tabElements.indexOf(activeElement);
+      },
+      getIndexOfTab: (tabToFind: Tab) => this._getTabs().indexOf(tabToFind),
+      getTabListLength: () => this._getTabs().length,
+      notifyTabActivated: (index: number) => {
+        // Synchronize the tabs `activeIndex` to the foundation.
+        // This is needed when a tab is changed via a click, for example.
+        this.activeIndex = index;
+        this.dispatchEvent(
+          new CustomEvent(MDCTabBarFoundation.strings.TAB_ACTIVATED_EVENT,
+            {detail: {index}, bubbles: true, cancelable: true}))
```

Comment:
do events like this need to be `composed: true` too? they will stop at the nearest shadow boundary otherwise (though you may want that?)